src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -60,8 +54,7 @@ public async Task Initialize()
                 return;
             }
 
-            var mainPipeline = new Pipeline<ITransportReceiveContext>(builder, pipelineConfiguration.Modifications);
-            mainPipelineExecutor = new MainPipelineExecutor(builder, eventAggregator, pipelineCache, mainPipeline, messageMapper);
+            mainPipelineExecutor = new MainPipelineExecutor(builder, pipeline);
```

Comment:
imo this should be part of the PipelineComponent as well, ReceiveComponent should not be the place to setup the MainPipelineExecutor